### DISCUSSION TITLE
Feature request: port NTP server functionality from GPS driver to the DS3231 driver.

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#11955] [Ideas]
Feature request: port NTP server functionality from GPS driver to the DS3231 driver. 

In remote locations, internet access and even access to another device with a RTC can be limited. As a limited solution, DS3231 provides reasonably accurate timekeeping and could be used to sync all devices on a remote network. Occasional passes with a cell phone with wifi sharing could re-sync the DS3231 and subsequently all devices on the subnet automatically, vs having to access/update each device.

**Comment by barbudor** [comment]
I would assume by your description that you are not running any MQTT server ?

I have a small location where Internet access is a problem but still have a raspberrypi running MQTT+Node-red. I've installed a small NTP server on the Pi (not the full ntpd but just a small relay) and it works fine.

**Reply by pkkrusty** [reply]
I am running a central pi with MQTT, syslog, and NTP, which is acting as my stratum 1 server since it has an attached GPS module. That all works great when the internal network is functional. But when that's down I have multiple remote sub-sites that have two or three devices that need a RTC reference. 

I could get a DS3231 for each device, or a GPS module for each sub-site. 

But a single DS3231 for each sub-site would mean less cost, and no need to worry about satellite visibility. 

I'm working on a 100 hectare farm in rural Senegal, with buildings spread over the whole area, all linked with Wifi antennas, but during rainy season (wifi signal issues) and solar grid failures, I'd like a way to keep time at each building without relying on the central server.

**Reply by pkkrusty** [reply]
I like the idea that my $2.25 D1 Mini and my $2.25 DS3231 could be a NTP server, vs $7.35 for the D1 Mini and NEO-6MV2 GPS module. I could save $2!

**Reply by barbudor** [reply]
Thanks for giving details of your use case 👍 
I also see cases when it may not be possible to use a GPS due to indoor reception for example;
It could make quite some sense.

I would expect that if someone pickup the idea and implement it the PR would be accepted

**Reply by pkkrusty** [reply]
Yeah, I wish I had the programming chops to do it myself.
